src/perf.js

```diff
@@ -185,117 +186,10 @@ router
-    const runs = {};
-    const transform = ({ graph }, start = null, end = null) => {
-      return graph.timelist
-        .map((date, idx) => {
-          const values = {
-            date: date * 1000,
-          };
-          graph.lines.forEach((line, lineIdx) => {
-            if (line && line.data[idx]) {
-              runs[lineIdx] = line.data[idx][0];
-            }
-          }, []);
-          if (runs[0] && (runs[1] || runs[2])) {
-            values.diff = ((runs[1] || runs[2]) - runs[0]) / runs[0] * 100;
-          }
-          return values;
-        })
-        .filter(entry => entry.diff)
-        .filter(entry => !end || entry.date < end)
-        .filter(entry => !start || entry.date > start);
-    };
-    const reference = transform(referenceSeries);
-    ctx.body = reference;
-  })
-  .get('/benchmark/speedometerBeta', async (ctx) => {
-    // idx 4 == beta
-    // idx 0 == chrome
-    const referenceSeries = await fetchJson('https://arewefastyet.com/data.php?file=aggregate-speedometer-misc-36.json');
-    const runs = {};
-    const transform = ({ graph }, start = null, end = null) => {
-      return graph.timelist
-        .map((date, idx) => {
-          const values = {
-            date: date * 1000,
-          };
-          graph.lines.forEach((line, lineIdx) => {
-            if (line && line.data[idx]) {
-              runs[lineIdx] = line.data[idx][0];
-            }
-          }, []);
-          if (runs[0] && runs[4]) {
-            values.diff = (runs[4] - runs[0]) / runs[0] * 100;
-          }
-          return values;
-        })
-        .filter(entry => entry.diff)
-        .filter(entry => !end || entry.date < end)
-        .filter(entry => !start || entry.date > start);
-    };
-    const reference = transform(referenceSeries);
-    ctx.body = reference;
-  })
-  .get('/benchmark/speedometerBeta32', async (ctx) => {
-    // idx 2 == beta
-    // idx 0 == chrome
-    const referenceSeries = await fetchJson('https://arewefastyet.com/data.php?file=aggregate-speedometer-misc-37.json');
-    const runs = {};
-    const transform = ({ graph }, start = null, end = null) => {
-      return graph.timelist
-        .map((date, idx) => {
-          const values = {
-            date: date * 1000,
-          };
-          graph.lines.forEach((line, lineIdx) => {
-            if (line && line.data[idx]) {
-              runs[lineIdx] = line.data[idx][0];
-            }
-          }, []);
-          if (runs[0] && runs[2]) {
-            values.diff = (runs[2] - runs[0]) / runs[0] * 100;
-          }
-          return values;
-        })
-        .filter(entry => entry.diff)
-        .filter(entry => !end || entry.date < end)
-        .filter(entry => !start || entry.date > start);
-    };
-    const reference = transform(referenceSeries);
-    ctx.body = reference;
+    if (Object.keys(ctx.request.query).length === 0) {
+      ctx.throw(400, 'The API now requires some parameters to be passed');
```

Comment:
It looks like you're looking for some specific parameters, maybe check for those explicitly and let the user know what they are?